### PR TITLE
feat(db): RHICOMPL-1561 make account numbers unique

### PR DESCRIPTION
**Pull request by skateman**
:thinking: not sure how to test this as the migration runs first and will not allow me to create non-unique accounts :thinking: we should reconsider to use rspec for migration tests that would support testing the up/down calls directly

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: db/migrate/20210809071351_deduplicate_accounts.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class DeduplicateAccounts < ActiveRecord::Migration[5.2]
+  def up
+    Account.transaction do
+      DuplicateAccountResolver.run!
+      remove_index :accounts, :account_number
+      add_index :accounts, :account_number, unique: true
+    end
```
Comment (akofink) [inline db/migrate/20210809071351_deduplicate_accounts.rb:7]: No need for an explicit transaction. From https://edgeguides.rubyonrails.org/active_record_migrations.html#migration-overview:

> On databases that support transactions with statements that change the schema, migrations are wrapped in a transaction. If the database does not support this then when a migration fails the parts of it that succeeded will not be rolled back. You will have to rollback the changes that were made by hand.

**Comment by akofink** [comment]
> thinking not sure how to test this as the migration runs first and will not allow me to create non-unique accounts thinking we should reconsider to use rspec for migration tests that would support testing the up/down calls directly

Wouldn't something like this work? https://github.com/RedHatInsights/compliance-backend/blob/master/test/services/duplicate_benchmark_resolver_test.rb#L12

**Review by vkrizan — APPROVED** [review]
LGTM. I recommend to do a thorough testing with some data in some pre-prod env.

**Comment by vkrizan** [comment]
@skateman also please update `schema.rb`. Thanks.

File: db/migrate/20210809071351_deduplicate_accounts.rb (hunk 1 above)
Comment (skateman) [reply db/migrate/20210809071351_deduplicate_accounts.rb:7]: Oh, :facepalm: okay, it will :sweat_smile:

**Review by akofink — COMMENTED** [review]
Thank you for the test! Still a redundant transaction in the migration, but other than that, it looks good. :+1: We should try to set up CI or QA at least with some duplicates before merging.

**Comment by skateman** [comment]
@akofink can you be more specific?

File: db/migrate/20210809071351_deduplicate_accounts.rb (hunk 1 above)
Comment (akofink) [reply db/migrate/20210809071351_deduplicate_accounts.rb:7]: @skateman maybe you forgot to push this update?

**Review by akofink — APPROVED** [review]
Thanks! :+1:

**Pull request merged**